rero_ils/modules/loans/api.py

```diff
@@ -290,6 +290,122 @@ def date_fields2str(self):
+            """Get pending loans."""
+            return LoansSearch()\
+                .params(preserve_order=True)\
+                .filter('term', state=LoanState.PENDING)\
+                .filter('term', library_pid=library_pid)\
+                .sort({'_created': {"order": 'asc'}})\
+                .source(includes=[
+                    'pid', 'transaction_date', 'item_pid', 'patron_pid',
+                    'document_pid', 'library_pid', 'state',
+                    'transaction_location_pid', 'pickup_location_pid'
+                ])\
+                .scan()
+
+        def patron_by_pid(patron_pid):
+            """Get patron by pid."""
+            if patron_pid not in patrons:
+                patrons[patron_pid] = PatronsSearch()\
+                    .filter('term', pid=patron_pid)\
+                    .source(includes=[
+                        'pid', 'first_name', 'last_name', 'patron.barcode'])\
+                    .execute()[0].to_dict()
+            return patrons[patron_pid]
+
+        def location_by_pid(location_pid):
+            """Get location by pid."""
+            if location_pid not in locations:
+                locations[location_pid] = LocationsSearch()\
+                    .filter('term', pid=location_pid)\
+                    .source(includes=['pid', 'name', 'library'])\
+                    .execute()[0].to_dict()
+            return locations[location_pid]
+
+        def library_name_by_pid(library_pid):
+            """Get library name by pid."""
+            if library_pid not in libraries:
+                libraries[library_pid] = LibrariesSearch()\
+                    .filter('term', pid=library_pid)\
+                    .source(includes='name')\
+                    .execute()[0].to_dict()['name']
+            return libraries[library_pid]
+
+        def item_by_pid(item_pid):
+            """Get item by pid."""
+            if item_pid not in items:
+                items[item_pid] = ItemsSearch()\
+                    .filter('term', pid=item_pid)\
+                    .filter('term', status=ItemStatus.ON_SHELF)\
+                    .source(includes=[
+                        'pid', 'barcode', 'call_number', 'library',
+                        'location', 'temporary_item_type'
+                    ]).execute()
+            return items[item_pid]
+
+        def item_type_by_pid(item_type_pid):
+            """Get item type by pid."""
+            if item_type_pid not in item_types:
+                item_types[item_type_pid] = ItemTypesSearch()\
+                    .filter('term', pid=item_type_pid)\
+                    .filter('term', negative_availability=True)\
+                    .execute()
+            return item_types[item_type_pid]
+
+        metadata = []
+        item_pids = []
+
+        loans = loans_pending()
+        for loan in loans:
+            item_pid = loan['item_pid']['value']
+            item = item_by_pid(item_pid)
+            if (len(item) > 0):
```

Comment:
`item_by_pid` seems return an Item not a list. This test seems invalid. Try to use `if item:`
Not necessary to enclose condition between parentheses in python (best practice).